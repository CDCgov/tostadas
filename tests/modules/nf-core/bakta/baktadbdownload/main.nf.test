nextflow_process {

    name "Test Process BAKTA_BAKTADBDOWNLOAD"
    script "modules/nf-core/bakta/baktadbdownload/main.nf"
    process "BAKTA_BAKTADBDOWNLOAD"

    test("Bakta database download") {

        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.db.get(0)).exists() },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

    test("Bakta database download - stub") {
    
        options "-stub"

        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.db + 
                    process.out.versions
                    ).match() }
            )
        }

    }

}
