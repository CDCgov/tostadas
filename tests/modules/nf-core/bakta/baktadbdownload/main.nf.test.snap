{
    "Bakta database download": {
        "content": [
            [
                "versions.yml:md5,29d6ec77dc88492b2c53141e6541c289"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:25:22.893231628"
    },
    "Bakta database download - stub": {
        "content": [
            [
                [
                    
                ],
                "versions.yml:md5,29d6ec77dc88492b2c53141e6541c289"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:25:28.780679353"
    }
}